### PR TITLE
Plugin updates

### PR DESCRIPTION
**Pull request by rmulhol**
- Emit specific errors with custom errors when execution fails
- Put tmp directories inside vdb directory (so we don't need permissions outside of it for execution)
- Call goose via library rather than binary (so we don't need the binary in the execution environment)

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -123,8 +123,8 @@ func initConfig() {
 		log.Printf("Using config file: %s\n\n", viper.ConfigFileUsed())
 	} else {
 		invalidConfigError := "Couldn't read config file"
-		fmt.Println("Error: ", invalidConfigError)
-		log.Fatal(invalidConfigError)
+		formattedError := fmt.Sprintf("%s: %s", invalidConfigError, err.Error())
+		log.Fatal(formattedError)
 		os.Exit(1)
```
Comment (yaoandrew) [inline cmd/root.go]: Not a big deal, but log.Fatal calls os.Exit(1) so you can lose this line.

Comment (rmulhol) [reply cmd/root.go]: I'll just rebase on your PR :D

File: pkg/plugin/manager/manager.go [hunk 2]
```diff
@@ -73,17 +92,19 @@ func (m *manager) RunMigrations() error {
...
+		removeErrString := "unable to remove file found at %s where tmp directory needs to be written: %s"
+		return errors.New(fmt.Sprintf(removeErrString, m.tmpMigDir, removeErr.Error()))
 	}
-	err = os.Mkdir(m.tmpMigDir, os.FileMode(os.ModePerm))
-	if err != nil {
-		return errors.New(fmt.Sprintf("unable to create temporary migration directory %s", m.tmpMigDir))
+	mkdirErr := os.Mkdir(m.tmpMigDir, os.FileMode(os.ModePerm))
+	if mkdirErr != nil {
+		mkdirErrString := "unable to create temporary migration directory %s: %s"
+		return errors.New(fmt.Sprintf(mkdirErrString, m.tmpMigDir, mkdirErr.Error()))
```
Comment (m0ar) [inline pkg/plugin/manager/manager.go]: `fmt.Errorf` fyi  :)

**Comment by m0ar** [comment]
@rmulhol Late to the party, but great fixes! Especially nice with more expressive error messages :pray:

**Pull request merged**